business/proxy_logging.go

```diff
@@ -0,0 +1,15 @@
+package business
+
+import (
+	"github.com/kiali/kiali/kubernetes"
+)
+
+// ProxyLoggingService is a thin layer over the kube interface for proxy logging functions.
+type ProxyLoggingService struct {
+	k8s kubernetes.ClientInterface
+}
+
+// SetLogLevel sets the pod's proxy log level.
+func (in *ProxyLoggingService) SetLogLevel(namespace, pod, level string) error {
+	return in.k8s.SetProxyLogLevel(namespace, pod, level)
```

Comment:
Suggestion, perhaps in the business layer you may want to check if the pod is in the mesh before to trying to perform the change.

This info is cached and you could re-use a single call to "kialiCache.GetPodProxyStatus(ns, pod)" (similar to used in the business/proxy_status.go) to let you perform that operation.

Non a blocker, you can rely on the runtime error of the "SetProxyLogLevel", but I was wondering that you may have that info available to decide if to throw the operation.

Reply:
If the pod is not in the mesh, would `GetPodProxyStatus` just return the runtime error returned by doing `curl localhost:15000/debug/` through the proxy and failing? That seems the same as just returning the runtime error from `SetProxyLogLevel`?

Reply:
Although it would probably be better to fail on doing a `GET` request to a port in the pod that is potentially running something else if it's not running the proxy rather than failing on a `POST` that could change something. Chances are small that the pod is running another service on `localhost:15000` that would be altered in some way from the request but not zero.

I'll update by doing that pod proxy status check first as you suggested.